test/e2e/common/04_tektonchainsgettingstartedtutorial_test.go

```diff
@@ -0,0 +1,297 @@
+		TektonChain:     "chain",
+		TargetNamespace: "tekton-pipelines",
+	}
+
+	if os.Getenv("TARGET") == "openshift" {
+		crNames.TargetNamespace = "openshift-pipelines"
+	}
+
+	clients := client.Setup(t, crNames.TargetNamespace)
+
+	utils.CleanupOnInterrupt(func() { utils.TearDownPipeline(clients, crNames.TektonPipeline) })
+	defer utils.TearDownPipeline(clients, crNames.TektonPipeline)
+
+	utils.CleanupOnInterrupt(func() { utils.TearDownChain(clients, crNames.TektonChain) })
+	defer utils.TearDownChain(clients, crNames.TektonChain)
+
+	resources.EnsureNoTektonConfigInstance(t, clients, crNames)
+
+	// Create a TektonPipeline
+	if _, err := resources.EnsureTektonPipelineExists(clients.TektonPipeline(), crNames); err != nil {
+		t.Fatalf("TektonPipeline %q failed to create: %v", crNames.TektonPipeline, err)
+	}
+
+	// Test if TektonPipeline can reach the READY status
+	t.Run("create-pipeline", func(t *testing.T) {
+		resources.AssertTektonPipelineCRReadyStatus(t, clients, crNames)
+	})
+
+	// Create a TektonChain
+	if _, err := resources.EnsureTektonChainExists(clients.TektonChains(), crNames); err != nil {
+		t.Fatalf("TektonChain %q failed to create: %v", crNames.TektonChain, err)
+	}
+
+	// Test if TektonChain can reach the READY status
+	t.Run("create-chain", func(t *testing.T) {
+		resources.AssertTektonChainCRReadyStatus(t, clients, crNames)
+	})
+
+	cosignSecret := "signing-secrets"
+	chainsConfigMapName := "chains-config"
+
+	// cosign generate-key-pair k8s://tekton-chains/signing-secrets
+	t.Run("create cosign key pair", func(t *testing.T) {
+		err := resources.CosignGenerateKeyPair(crNames.TargetNamespace, cosignSecret)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	// kubectl patch configmap chains-config -n tekton-chains -p='{"data":{"artifacts.oci.storage": "", "artifacts.taskrun.format":"tekton", "artifacts.taskrun.storage": "tekton"}}'
+	chainsConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      chainsConfigMapName,
+			Namespace: crNames.TargetNamespace,
+		},
+		Data: map[string]string{
+			"artifacts.oci.storage":     "",
+			"artifacts.taskrun.format":  "tekton",
+			"artifacts.taskrun.storage": "tekton",
+		},
+	}
+
+	t.Run("replace Chains ConfigMap", func(t *testing.T) {
+		_, err := resources.ReplaceConfigMap(clients.KubeClient, chainsConfigMap)
+		if err != nil {
+			t.Fatal(err)
+		}
+	})
+
+	// kubectl delete po -n tekton-chains -l app=tekton-chains-controller
+	t.Run("restart chains pod", func(t *testing.T) {
+		err := resources.DeleteChainsPod(clients.KubeClient, crNames.TargetNamespace)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// make sure the new pod is up and running again
+		resources.AssertTektonChainCRReadyStatus(t, clients, crNames)
+	})
+
+	// kubectl create -f https://raw.githubusercontent.com/tektoncd/chains/main/examples/taskruns/task-output-image.yaml
+	taskRunName := "build-push-run-output-image-test"
+	taskOutputImageTaskRun := v1alpha1.TaskRun{
```

Comment:
We could use the [`github.com/tektoncd/pipeline/test/parse`](https://github.com/tektoncd/pipeline/blob/main/test/parse/yaml.go) package here to be able to inline the yaml instead of writing the struct as a go object, it might render the test more readable.

Reply:
great suggestion, fixed